e2e/ansible/playbooks/hyperconverged/test-scaleup-cassandra-pods/test-scaleup-cassandra-pods.yml

```diff
@@ -0,0 +1,231 @@
+- hosts: localhost
+ 
+  vars_files: 
+    - scaleup-vars.yml 
+ 
+  tasks:
+   - block:
+
+       - name: Get $HOME of K8s master for kubernetes user
+         shell: source {{ profile }}; echo $HOME
+         args: 
+           executable: /bin/bash
+         register: result_kube_home
+         delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+
+       - name: Download YAML for cassandra service
+         get_url: 
+           url: "{{ cassandra_service_link }}"
+           dest: "{{ result_kube_home.stdout }}/{{ svc_yaml_alias }}"
+           force: yes
+         register: result
+         until:  "'OK' in result.msg"
+         delay: 5
+         retries: 3
+         delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+
+       - name: Download YAML for cassandra statefulset
+         get_url:
+           url: "{{ cassandra_stateful_link }}"
+           dest: "{{ result_kube_home.stdout }}/{{ stateful_yaml_alias }}"
+           force: yes
+         register: result
+         until:  "'OK' in result.msg"
+         delay: 5
+         retries: 3
+         delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+
+       - name: Download YAML for cassandra loadgen
+         get_url:
+           url: "{{ cassandra_loadgen_link }}"
+           dest: "{{ result_kube_home.stdout }}/{{ loadgen_yaml_alias }}"
+           force: yes
+         register: result
+         until:  "'OK' in result.msg"
+         delay: 5
+         retries: 3
+         delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+   
+       - name: Get the number of nodes in the cluster
+         shell: source {{ profile }}; kubectl get nodes | grep 'Ready' | wc -l
+         args:
+           executable: /bin/bash
+         register: node_out
+         delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+
+       - name: Fetch the node count from stdout
+         set_fact:
+            node_count: " {{ node_out.stdout}}"
+
+
+       - name: Replace the replica count in cassandra statefulset yaml
+         replace:
+           path: "{{ result_kube_home.stdout }}/{{ stateful_yaml_alias }}"
+           regexp: 'replicas: 3'
+           replace: 'replicas: {{ (node_count) |int-2 }}'
```

Comment:
Can this step be made more intuitive. Lets say, the node count happens to be 2 !!